### PR TITLE
htlcswitch/mailbox_test: use mailboxContext everywhere

### PR DESCRIPTION
**Pull request by cfromknecht**
Modifies some of the older mailbox tests that were left untouched as of
PR #4174 to use the new mailbox test context. This ensures that the
config members of each mailbox's config are properly initialized. In
certain instances where travis is slow, this would cause test panics.

### CODE REVIEW

File: htlcswitch/mailbox_test.go [hunk 1]
```diff
@@ -20,12 +22,8 @@ func TestMailBoxCouriers(t *testing.T) {
 
 	// First, we'll create new instance of the current default mailbox
 	// type.
-	mailBox := newMemoryMailBox(&mailBoxConfig{
-		clock:  clock.NewDefaultClock(),
-		expiry: time.Minute,
-	})
-	mailBox.Start()
-	defer mailBox.Stop()
+	ctx := newMailboxContext(t, time.Now(), testExpiry)
```
Comment (bhandras) [inline htlcswitch/mailbox_test.go:25]: I'd recommend using `clock.TestClock` instead of real time here.

File: htlcswitch/mailbox_test.go [hunk 2]
```diff
@@ -163,18 +170,17 @@ func TestMailBoxCouriers(t *testing.T) {
 func TestMailBoxResetAfterShutdown(t *testing.T) {
 	t.Parallel()
 
-	m := newMemoryMailBox(&mailBoxConfig{})
-	m.Start()
+	ctx := newMailboxContext(t, time.Now(), time.Second)
```
Comment (bhandras) [inline htlcswitch/mailbox_test.go:173]: Same here and below: would be nice to use test clock instead.

File: htlcswitch/mailbox_test.go [hunk 3]
```diff
@@ -548,8 +548,20 @@ func TestMailOrchestrator(t *testing.T) {
...
+		fetchUpdate: func(sid lnwire.ShortChannelID) (
+			*lnwire.ChannelUpdate, error) {
+			return &lnwire.ChannelUpdate{
+				ShortChannelID: sid,
+			}, nil
+		},
+		forwardPackets: func(_ chan struct{},
+			pkts ...*htlcPacket) chan error {
+			errChan := make(chan error)
+			close(errChan)
```
Comment (bhandras) [inline htlcswitch/mailbox_test.go:560]: nit: Maybe add a comment explaining why the `errchan` is closed immediately.

File: htlcswitch/mailbox_test.go [hunk 4]
```diff
@@ -548,8 +548,20 @@ func TestMailOrchestrator(t *testing.T) {
...
+			return errChan
+		},
 		clock:  clock.NewDefaultClock(),
```
Comment (bhandras) [inline htlcswitch/mailbox_test.go:563]: We might as well change this to not depend on the `DefaultClock` but `TestClock` instead.

File: htlcswitch/mailbox_test.go (hunk 1 above)
Comment (cfromknecht) [reply htlcswitch/mailbox_test.go:25]: `newMailboxContext` already creates a `TestClock` internally ;)

File: htlcswitch/mailbox_test.go (hunk 4 above)
Comment (cfromknecht) [reply htlcswitch/mailbox_test.go]: done

File: htlcswitch/mailbox_test.go (hunk 3 above)
Comment (cfromknecht) [reply htlcswitch/mailbox_test.go:562]: done

File: htlcswitch/mailbox_test.go (hunk 1 above)
Comment (bhandras) [reply htlcswitch/mailbox_test.go:25]: Sneaky :) Somehow `time.Now()` made me think it does depend on actual time. Perhaps it's a good idea to fix the time anyway as then when we grep for `time.Now()` in tests we won't find this one. (Not a biggie though)

Comment (cfromknecht) [reply htlcswitch/mailbox_test.go:25]: > we grep for time.Now()

not sure i follow, why wouldn't this line be found? do you mean `clock.Now()`?

Comment (bhandras) [reply htlcswitch/mailbox_test.go:25]: There are a bunch of tests currently depending on `time.Now()` which is probably wrong and in the long run should be fixed to use `TestClock` instead. 
Just made a quick ` ag "time.Now()" | grep "_test.go" | wc -l`, seems like we have 144 occurrences.

**Pull request merged**